src/NuGetGallery/Controllers/ApiController.cs

```
@@ -289,15 +294,40 @@ public virtual Task<ActionResult> CreatePackagePost()
                         return new HttpStatusCodeResult(HttpStatusCode.Created);
                     }
                 }
+                catch (InvalidPackageException ex)
+                {
+                    return BadRequestForExceptionMessage(ex);
+                }
                 catch (InvalidDataException ex)
                 {
+                    return BadRequestForExceptionMessage(ex);
+                }
+                catch (EntityException ex)
+                {
+                    return BadRequestForExceptionMessage(ex);
+                }
+                catch (FrameworkException ex)
+                {
+                    return BadRequestForExceptionMessage(ex);
+                }
+                catch (Exception ex)
```

Are you sure you want to swallow unexpected exceptions? There might be error handlers down the line that will handle them better (like show an error page)